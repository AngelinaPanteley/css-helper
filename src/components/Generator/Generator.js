import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styles from './Generator.scss';

import Control from './Control/Control';

class Generator extends PureComponent {
  static propsTypes = {
    controls: PropTypes.object.isRequired,
    controlValues: PropTypes.object.isRequired,
    handleChange: PropTypes.func.isRequired,
    onClear: PropTypes.func.isRequired,
  }

  render() {
    const controls = [];
    for (let cont in this.props.controls) {
      const control = this.props.controls[cont];
      controls.push(<Control
        control={control}
        key={control.label}
        value={this.props.controlValues[cont]}
        handleChange={(value) => this.props.handleChange(cont, value)} />);
    }

    return (
      <div className={styles.Generator} >
        <button onClick={this.props.onClear}
          className={styles.ClearButton}>
          Clear All
        </button>
        {controls}
      </div>
    );
  }
}

export default Generator;