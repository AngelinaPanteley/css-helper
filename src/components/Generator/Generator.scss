@import '../../variables';

.ClearButton {
  background: none;
  color: $color-white;
  border: none;
  outline: none;
  font-size: 16px;
  padding: 0;
  margin: 20px 15px 10px;
  cursor: pointer;
  transition: $transition;
  min-height: 30px;

  &:hover,
  &:focus,
  &:active {
    transition: $transition;
    text-shadow: 0 0 1px $color-white;
  }
}

.Generator {
  @include flex-wrapper(column, flex-start, flex-end, nowrap);
  padding-bottom: 20px;
  max-height: calc(100vh - 128px);
}