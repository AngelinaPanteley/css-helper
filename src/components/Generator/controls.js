export const number = 'number';
export const color = 'color';