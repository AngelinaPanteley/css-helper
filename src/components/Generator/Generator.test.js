"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { expect as chaiExpect } from 'chai';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import { MemoryRouter } from "react-router-dom";
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import Generator from './Generator.js';
import reducer from '../../store/reducers/index';

configure({ adapter: new Adapter() });

const store = createStore(reducer);

describe('>>>Generator', () => {

  //SNAPSHOTS

  const props = {
    controls: {
      control: {
        type: 'number',
        initialValue: 30,
      },
    },
    controlValues: {
      control: 20,
    },
    onClear: () => { },
    handleChange: () => { },
  };

  const generatorComponent = <Provider store={store}><MemoryRouter>
    <Generator {...props} />
  </MemoryRouter></Provider>;

  const component = renderer.create(generatorComponent);

  it('renders control correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });

  //SHALLOW RENDERING

  let wrapper;
  const shallowComponent = <Generator {...props} />;

  beforeEach(() => {
    wrapper = shallow(shallowComponent);
  });

  it('renders clear button', () => {
    chaiExpect(wrapper.find('button')).to.have.lengthOf(1);
  });
});
