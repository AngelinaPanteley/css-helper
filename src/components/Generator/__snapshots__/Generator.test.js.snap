// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Generator renders control correctly 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Clear All
  </button>
  <div>
    <label>
      <span />
      <span>
        20
      </span>
    </label>
    <div
      className="rc-slider"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
        style={Object {}}
      />
      <div
        className="rc-slider-track"
        style={
          Object {
            "left": "0%",
            "width": "20%",
          }
        }
      />
      <div
        className="rc-slider-step"
      />
      <div
        aria-disabled={false}
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={20}
        className="rc-slider-handle"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="slider"
        style={
          Object {
            "left": "20%",
          }
        }
        tabIndex={0}
      />
      <div
        className="rc-slider-mark"
      />
    </div>
  </div>
</div>
`;
